File: src/org/baksia/rustycage/editors/RustEditor.java
package org.baksia.rustycage.editors;

import org.eclipse.ui.PlatformUI;
import org.eclipse.ui.editors.text.TextEditor;

public class RustEditor extends TextEditor {

    public RustEditor() {
        super();
//        addListenerObject();
        //TODO : chenge on focus gained and focus lost
        PlatformUI.getWorkbench().getThemeManager().setCurrentTheme("org.baksia.rustycage.ui.theme");
        setSourceViewerConfiguration(new RustConfiguration());
    }

    @Override
    public void dispose() {
        //TODO : Figure out defauøt theme and chenge on focus gained and focus lost
        PlatformUI.getWorkbench().getThemeManager().setCurrentTheme("Default");
        super.dispose();
    }

    @Override
    protected void createActions() {
        super.createActions();
    }

    @Override
    public void setFocus() {
        super.setFocus();
    }


}

File: src/org/baksia/rustycage/editors/RustEditor.scala
package org.baksia.rustycage.editors

import org.eclipse.ui.editors.text.TextEditor

class RustEditor extends TextEditor {

  setSourceViewerConfiguration(new RustConfiguration())
  
  override def dispose() {
    super.dispose()
  }
  
  override def createActions() {
    super.createActions()
  }

  override def setFocus() {
    super.setFocus()
  }
  
  
}
